plugins/analytics-module-nr/src/apis/implementations/AnalyticsApi/NewRelicBrowser.ts

```diff
@@ -0,0 +1,113 @@
+/*
+ * Copyright 2023 The Backstage Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import { Config } from '@backstage/config';
+import {
+  AnalyticsApi,
+  IdentityApi,
+  AnalyticsEvent,
+} from '@backstage/core-plugin-api';
+import { BrowserAgent } from '@newrelic/browser-agent/loaders/browser-agent';
+import type { setAPI } from '@newrelic/browser-agent/loaders/api/api';
+
+type NewRelicAPI = ReturnType<typeof setAPI>;
+
+type NewRelicBrowserOptions = {
+  accountId: string;
+  applicationId: string;
+  licenseKey: string;
+  distributedTracingEnabled: boolean;
+  cookiesEnabled: boolean;
+  useEuEndpoint: boolean;
+};
+
+/**
+ * New Relic Browser API provider for the Backstage Analytics API.
+ * @public
+ */
+export class NewRelicBrowser implements AnalyticsApi {
+  private readonly agent: NewRelicAPI;
+
+  private constructor(
+    options: NewRelicBrowserOptions,
+    identityApi?: IdentityApi,
+  ) {
+    // Configure the New Relic Browser agent
+    const agentOptions = {
+      init: {
+        distributed_tracing: {
+          enabled: options.distributedTracingEnabled,
+        },
+        privacy: {
+          cookies_enabled: options.cookiesEnabled,
+        },
+        ajax: {
+          deny_list: [
+            options.useEuEndpoint ? 'bam.eu01.nr-data.net' : 'bam.nr-data.net',
```

Comment:
Would it make more sense to change this to be just the endpoint in the config as maybe an enum and then you wouldn't need to this here and below in the `info` section? So instead of `useEuEndpoint` it would be `endpoint` and it would need to be one of `'bam.eu01.nr-data.net' | 'bam.nr-data.net'`

Reply:
I chose this as I'm not entirely sure that users would be familiar with those particular endpoints. They are documented of course on the New Relic side, but I think users would have an easier time to reason about either the US or the EU data centers, instead of specific endpoints. But I'd be happy to change it.

Reply:
@awanlin @benjdlambert I guess this is the only remaining issue here. Any thoughts on this?

Reply:
Still think it would look much nicer in code if you do something like this: 

`endpoint?: 'bam.eu01.nr-data.net' | 'bam.nr-data.net';`

An example: https://github.com/backstage/backstage/blob/c4a9eb45ab17668c3f04ac313f99d6430a7349f6/plugins/analytics-module-ga4/config.d.ts#L47

You could list the endpoints in the `README` and include a link to the relevant documentation. Also, maybe making this visible might help those with tight networking policies identify this as something they might need to open up? 

What do you think @benjdlambert ?

Reply:
Think this makes sense @awanlin :tada:

Reply:
Yeah, that makes sense. I've made these changes just now.